clang/lib/Headers/aiev2_core.h

```diff
@@ -1566,4 +1566,300 @@ inline v16accfloat addmsc_4x8_8x4( v32float v1, v32float v2, v16accfloat acc1, v
 
 #endif
 
+enum class aie_dm_resource {
+  none,
+  PM,
+  DM,
+  DM_test,
+  stack,
+  a,
+  b,
+  c,
+  d,
+  ab,
+  ac,
+  ad,
+  bc,
+  bd,
+  cd,
+  TM
+};
+
+#define __aie_dm_resource_PM                                                   \
```

Comment:
Should these #defines remain defined?

Reply:
How about:
```
#define FOO(x) __attribute__((address_space(static_cast<signed>(aie_dm_resource::x))))
#define __aie_dm_resource_DM FOO(DM)
#undef FOO
```
I find this much easier to see what the regular pattern is, especially given the awkward line breaks.